doc/train_parameters.md

```diff
@@ -0,0 +1,155 @@
+
+### COLUMN FOR
+The expressions in `COLUMN` contains the `Feature Columns` information. There could be multiple `COLUMN` blocks in the SQL. 
+
+The value of `FOR` keyword represents which parameter of the constructor method the `feature columns` assigning to.
+
+For example, the following SQL
+```sql
+...
+TRAIN
+	DNNLinearCombinedClassifier
+WITH
+	...
+COLUMN
+	NUMERIC(...) FOR linear_feature_columns
+COLUMN
+	BUCKET(...) FOR dnn_feature_columns
+...
+```
+will be translated to 
+```python
+	estimator = tf.estimator. DNNLinearCombinedClassifier(
+		linear_feature_columns = [tf.feature_column.numeric(...)],
+		dnn_feature_columns = [tf.feature_column.bucket(...)],
+		...
+	)
+```
+
+They were not only translated into [TensorFlow Feature Columns](https://www.tensorflow.org/guide/feature_columns), but also contains the encoding information of data such as `DENSE` or `SPARSE`. 
+
+For example, the following SQL
+```sql
+SELECT 
+...
+COLUMN
+	DENSE(c1, ...)
+	SPARSE(c2, ...)
+```
+represents that the `c1` field is the dense format and the `c2` field is the sparse format.
+
+<b>Here is a list of supported expression:</b>
+
+| expression | arguments of expression                                          | example                            |
+|------------|------------------------------------------------------------------|------------------------------------|
+| dense      | 1. field name (str) <br> 2. dense shape (integer) <br> 3. separator (str)  | dense(c1, 100, comma)              |
+| sparse     | 1. field name (str) <br> 2. sparse shape (integer) <br> 3. separator (str) | sparse(c2, 10000, comma)           |
+| numeric    | 1. field name (str) <br> 2. shape (integer)                           | numeric(c1, 100)                   |
+| bucket     | 1. key (numeric) <br> 2. bucket size (integer)                        | bucket(numeric(c1, 100), 20)       |
+| cat_id     | 1. field name (str) <br> 2. bucket size (integer)                     | cat_id(c2, 10000)                  |
+| embedding  | 1. key (cat_id) <br> 2. dimension (integer) <br> 3. combiner (str)         | embedding(cat_id(c2, 10000), mean) |
+
+
+## Independent module for resolving of training parameters
+Since the format of training parameters has been unified, it's better to have an independent module in SQLFlow to do the resolving according to the rules instead of doing it in each `codegen_**.go`.
+
+The advantage of an independent module contains
+1. Unifying resolving process to avoid differences in processing between `codegen` modules. 
+2. Fast fail during resolving before the code generation.
+
+Here we propose a module such as `resolving.go` which take the `trainClause` as input and outputs a struct named `resolvedTrainClause` for all kinds of `codegen_**.go` to do code generation.
+
+The struct named `resolvedTrainClause` looks like this
+```go
+package sql
+
+type resolvedTrainClause struct {
+	IsPreMadeModel  bool
```

Comment:
We would also like to support keras models, and the generated code could have a lot of difference than estimator code. Will the "resolver" also consider dealing with keras models?